tests/registration/03-skip-run.t

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2016 NIWA
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#------------------------------------------------------------------------------
+# Test skip unregister if suite has a port file.
+
+. "$(dirname "$0")/test_header"
+set_test_number 6
+
+CYLC_RUND="$(cylc get-global-config --print-run-dir)"
+
+SUITED_1="$(mktemp -d --tmpdir="${CYLC_RUND}" 'ctb-registration-03-1XXXXXXXX')"
+SUITED_2="$(mktemp -d --tmpdir="${CYLC_RUND}" 'ctb-registration-03-2XXXXXXXX')"
+SUITED_R="$(mktemp -d --tmpdir="${CYLC_RUND}" 'ctb-registration-03-RXXXXXXXX')"
+
+for SUITED in "${SUITED_1}" "${SUITED_2}" "${SUITED_R}"; do
+    cat >"${SUITED}/suite.rc" <<'__SUITERC__'
+[cylc]
+    [[event hooks]]
+        timeout = PT1M
+        abort on timeout = True
+[scheduling]
+    [[dependencies]]
+        graph = foo
+[runtime]
+    [[foo]]
+        script = printenv FOO
```

Comment:
Ok - to make the task fail, but `/bin/false` would be more obvious (no need to check if `$FOO` is defined anywhere).